### PR TITLE
Rename .ml-state index to .ml-state-000001 to support rollover

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR changes the name of the state index from `.ml-state` to `.ml-state-000001` so that the rollover can be performed on the index.
The state index pattern (`.ml-state*`) is still valid as it "catches" both legacy and new name.
If the user already has the legacy state index (`.ml-state`), we will not create the new one (`.ml-state-000001`). 

This PR doesn't create any ILM policy nor does it invoke the rollover action. It can be viewed as a prerequisite for implementing either ILM or rollover. 

Relates https://github.com/elastic/elasticsearch/issues/29938

### CODE REVIEW

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/30_ml_jobs_crud.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+setup:
+  - do:
+      indices.create:
+        index: .ml-state
```
Comment (droberts195) [inline x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/30_ml_jobs_crud.yml]: Please can you create this with realistic mappings and settings.  Something like:

```
  "settings" : {
    "index" : {
      "auto_expand_replicas" : "0-1"
    }
  },
  "mappings" : {
    "_doc": {
      "_meta": {
        "version": "7.0.0"
      },
      "enabled": false
    }
  }
```

**Review by droberts195 — APPROVED** [review]
LGTM if you could just add the realistic settings and mappings (and of course get the tests to pass)

File: x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/30_ml_jobs_crud.yml (hunk 1 above)
Comment (przemekwitek) [reply x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/30_ml_jobs_crud.yml]: Aren't these settings going to be inherited from the template?
`x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/anomalydetection/state_index_template.json`
Let me verify that.

Comment (droberts195) [reply x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/30_ml_jobs_crud.yml]: Oh yes, good point

Comment (przemekwitek) [reply x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/30_ml_jobs_crud.yml]: > Aren't these settings going to be inherited from the template?
x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/anomalydetection/state_index_template.json
Let me verify that.

Confirmed, the settings are picked up from the template.
Now, I'm going to make these tests pass.

**Pull request merged**